odl/discr/grid.py

```diff
@@ -515,6 +515,40 @@ def insert(self, index, other):
                     self.coord_vectors[idx:])
         return TensorGrid(*new_vecs)
 
+    def remove(self, axes):
+        """Remove ``axes`` and return the reduced grid.
+
+        Note that no changes are made in-place.
+
+        Parameters
+        ----------
+        axes : index expression
```

Comment:
This indicates that an arbitrary index expression works, but I'm quite sure this only works with integers or lists of integers.